### PR TITLE
5 persist and use local PERF in Epitelete

### PR DESCRIPTION
**Pull request by abelpz**

### CODE REVIEW

**Review by mvahowe — CHANGES_REQUESTED** [review]
This looks good - I think all the requested functionality is there, and the tests are extensive and clear.

I made some minor changes to make the async testing of throwing clearer using `t.pass`. If you do `npm run rawTest` you'll see that this gives you, eg `ok 5 fetchPerf throws on bad bookCode
` rather than a random error.

In general I don't write tests for if a method exists, on the assumption that, if it doesn't, every other test will break anyway. But people do that in a pure test-driven dev paradigm, and it doesn't do any harm, and might even do some good, so there's no need to change anything.

The thing to change is flattening the tests. If you look at the `npm test` output you'll see

```✔ fetchPerf method (Smoke) [pass: 0, fail: 0, duration: 1ms]```

In other words, I don't think that nesting tests is how you structure tests in tape. (I think it might be how you do things with some other test frameworks). The pattern I use in tape is
- a flat series of test()
- each test may contain quite a lot of assertions

For now, we just need to remove the outer `test()` functions so all the remaining `test()` functions are at the top level. Another time, I would consider putting the assertions of several of those functions inside a single `test()` so, eg, you check that the function exists and that it throws or doesn't throw and that the basic use case works inside a single test, the results of which will then be aggregated by `npm test`. But there's no need to do that here.

File: test/code/smoke.cjs [hunk 1]
```diff
@@ -27,38 +27,196 @@ test(
...
+                t.end()
+            }
+        )
+
+        t.test(
+            `fetchPerf() adds document to documents property`,
+            async t => {
+                t.plan(2);
+                const bookCode = "MRK";
+                t.notOk(bookCode in epitelete.documents);
```
Comment (mvahowe) [inline test/code/smoke.cjs:100]: This is great! Check that the thing you want to happen hasn't already happened before you run the code!

**Review by mvahowe — APPROVED** [review]
Thanks for the fixes, looks perfect!
